conda_libmamba_solver/shards_cache.py

```diff
@@ -61,16 +61,28 @@ class ShardCache:
     Handle caching for individual shards (not the index of shards).
     """
 
-    def __init__(self, base: Path):
+    def __init__(self, base: Path, create=True):
         """
         base: directory and filename prefix for cache.
         """
         self.base = base
         self.connect()
 
-    def connect(self):
+    def copy(self):
+        """
+        Copy cache with new connection. Useful for threads.
+        """
+        return ShardCache(self.base, create=False)
```

Comment:
`sqlite3.OperationalError: disk I/O error` appeared in the thread, avoid redundant 'create table if not exists' in `copy()`.